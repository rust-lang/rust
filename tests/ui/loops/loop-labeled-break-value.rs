fn main() {
    loop {
        let _: i32 = loop { break }; //~ ERROR mismatched types
    }
    loop {
        let _: i32 = 'inner: loop { break 'inner }; //~ ERROR mismatched types
    }
    loop {
        let _: i32 = 'inner2: loop { loop { break 'inner2 } }; //~ ERROR mismatched types
    }
}
