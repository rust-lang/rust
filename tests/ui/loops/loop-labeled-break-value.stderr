error[E0308]: mismatched types
  --> $DIR/loop-labeled-break-value.rs:3:29
   |
LL |         let _: i32 = loop { break };
   |             -        ----   ^^^^^ expected `i32`, found `()`
   |             |        |
   |             |        this loop is expected to be of type `i32`
   |             expected because of this assignment
   |
help: give the `break` a value of the expected type
   |
LL |         let _: i32 = loop { break 42 };
   |                                   ++

error[E0308]: mismatched types
  --> $DIR/loop-labeled-break-value.rs:6:37
   |
LL |         let _: i32 = 'inner: loop { break 'inner };
   |             -                ----   ^^^^^^^^^^^^ expected `i32`, found `()`
   |             |                |
   |             |                this loop is expected to be of type `i32`
   |             expected because of this assignment
   |
help: give the `break` a value of the expected type
   |
LL |         let _: i32 = 'inner: loop { break 'inner 42 };
   |                                                  ++

error[E0308]: mismatched types
  --> $DIR/loop-labeled-break-value.rs:9:45
   |
LL |         let _: i32 = 'inner2: loop { loop { break 'inner2 } };
   |             -                 ----          ^^^^^^^^^^^^^ expected `i32`, found `()`
   |             |                 |
   |             |                 this loop is expected to be of type `i32`
   |             expected because of this assignment
   |
help: give the `break` a value of the expected type
   |
LL |         let _: i32 = 'inner2: loop { loop { break 'inner2 42 } };
   |                                                           ++

error: aborting due to 3 previous errors

For more information about this error, try `rustc --explain E0308`.
